File: data/code_pairs/fortran/1909.f90
program chk_alloc_string
    implicit none
    character(len=:), dimension(:), allocatable :: strings
    character(20) :: buffer
    integer :: i

    allocate(character(len=10) :: strings(5))
    strings = ["A   ", "C   ", "ABCD", "V   ", "zzzz"]

    ! The checks are embedded within the test program below.
    ! The output formatting and checks would typically occur here or in a subroutine.

    write(buffer, "(5a4)") strings
    print *, buffer
end program chk_alloc_string

File: data/code_pairs/cpp/1909.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <algorithm>

std::string formatStrings(const std::vector<std::string>& strings) {
    std::ostringstream buffer;
    for (const auto& str : strings) {
        buffer.width(4);
        buffer << std::left << str.substr(0, 4);
    }
    return buffer.str();
}

int main() {
    std::vector<std::string> strings = {"A", "C", "ABCD", "V", "zzzz"};
    std::string formattedOutput = formatStrings(strings);

    std::cout << formattedOutput << std::endl;

    return 0;
}